pkg/runtime/indexer/indexer_test.go

```diff
@@ -0,0 +1,103 @@
+package indexer
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	trainer "github.com/kubeflow/trainer/pkg/apis/trainer/v1alpha1"
+	utiltesting "github.com/kubeflow/trainer/pkg/util/testing"
+)
+
+func TestIndexTrainJobTrainingRuntime(t *testing.T) {
+	cases := map[string]struct {
+		obj  client.Object
+		want []string
+	}{
+		"object is not a TrainJob": {
+			obj: utiltesting.MakeTrainingRuntimeWrapper(metav1.NamespaceDefault, "test").Obj(),
+		},
+		"TrainJob with matching APIGroup and Kind": {
+			obj: utiltesting.MakeTrainJobWrapper(metav1.NamespaceDefault, "test").
+				RuntimeRef(trainer.GroupVersion.WithKind(trainer.TrainingRuntimeKind), "test runtime").Obj(),
```

Comment:
```suggestion
				RuntimeRef(trainer.GroupVersion.WithKind(trainer.TrainingRuntimeKind), "runtime").Obj(),
```

We could avoid using an empty space.
Please address this comment in other places as well, thanks.